data/waveform_generator.js

```diff
@@ -0,0 +1,147 @@
+let Stream = require('./base_stream.js')
+let Utils = require('./utils.js')
+
```

Comment:
Add description of this generator (general - how it works, etc.) in front of the constructor.